Sources/MongoSwift/MongoCollection+BulkWrite.swift

```diff
@@ -81,12 +89,16 @@ extension MongoCollection {
         }
 
         /// Adds the `deleteMany` operation to a bulk write
+        ///
+        /// - Throws:
+        ///   - `UserError.invalidArgumentError` if the options form an invalid combination.
+        ///   - `EncodingError` if an error occurs while encoding the options to BSON.
         public func addToBulkWrite(bulk: BulkWriteOperation, index: Int) throws {
             let opts = try BSONEncoder().encode(options)
             var error = bson_error_t()
 
             guard mongoc_bulk_operation_remove_many_with_opts(bulk.bulk, self.filter.data, opts.data, &error) else {
-                throw MongoError.invalidArgument(message: toErrorString(error))
+                throw UserError.invalidArgumentError(message: toErrorString(error))
```

Comment:
Why isn't this line using `parseMongocError`?  It seems like there are a few more cases like this, I'll restrict my comment to this one